internal/buildsign/ocpbuild/ocpbuildmanager.go

```diff
@@ -0,0 +1,173 @@
+package ocpbuild
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	buildv1 "github.com/openshift/api/build/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/rh-ecosystem-edge/kernel-module-management/internal/constants"
+)
+
+type Status string
+
+const (
+	ocpbuildTypeBuild = "build"
+	ocpbuildTypeSign  = "sign"
+
+	StatusCompleted  Status = "completed"
+	StatusCreated    Status = "created"
+	StatusInProgress Status = "in progress"
+	StatusFailed     Status = "failed"
+
+	hashAnnotation = "kmm.node.kubernetes.io/last-hash"
+)
+
+var ErrNoMatchingBuild = errors.New("no matching Build")
+
+//go:generate mockgen -source=ocpbuildmanager.go -package=ocpbuild -destination=mock_ocbuildmanager.go
+
+type ocpbuildManager interface {
+	isOCPBuildChanged(existingBuild *buildv1.Build, newBuild *buildv1.Build) (bool, error)
+	getModuleOCPBuildByKernel(ctx context.Context, modName, namespace, kernelVersion, ocbuildType string, owner metav1.Object) (*buildv1.Build, error)
+	getModuleOCPBuilds(ctx context.Context, modName, modNamespace, ocbuildType string, owner metav1.Object) ([]buildv1.Build, error)
+	getOCPBuildStatus(build *buildv1.Build) (Status, error)
+	createOCPBuild(ctx context.Context, build *buildv1.Build) error
+	deleteOCPBuild(ctx context.Context, build *buildv1.Build) error
+	ocpbuildLabels(modName, kernelVersion, ocpbuildType string) map[string]string
```

Comment:
```suggestion
	getOCPBuildLabels(modName, kernelVersion, ocpbuildType string) map[string]string
```

Reply:
it is not a set operation, it  just returns the labels. In the upstream we have a similar function, which is called PodLabels